src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,73 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+
+        return storageDao.add(user);
+    }
+
+    private void validateLoginUniqueness(String login) {
+        if (storageDao.get(login) != null) {
+            throw new UserDataInvalidException("User with such login already exists");
+        }
+    }
+
+    private void validateLogin(String login) {
+        if (login.length() < MIN_LOGIN_LENGTH) {
+            throw new UserDataInvalidException("User`s login length should contain at least"
+                    + MIN_LOGIN_LENGTH
+                    + "characters");
+        }
+    }
+
+    private void validateAge(Integer age) {
+        if (age < MIN_USER_AGE) {
+            throw new UserDataInvalidException("User should be at least "
+                    + MIN_USER_AGE
+                    + " years old");
+        }
+    }
+
+    private void validatePassword(String password) {
+        if (password.length() < MIN_PASSWORD_LENGTH) {
+            throw new UserDataInvalidException("User`s password length should contain at least "
+                    + MIN_PASSWORD_LENGTH
+                    + " characters");
+        }
+    }
+
+    private void checkNotNull(Object value, String errorMessage) {
+        if (value == null) {
+            throw new UserDataInvalidException(errorMessage);
```

Comment:
While the `checkNotNull` method is a good way to avoid code duplication, it's not checking for empty strings which might be important for login and password fields. Consider adding checks for empty strings in the `validateLogin` and `validatePassword` methods.

Reply:
Length is checked by other methods